src/platform/forms-system/src/js/state/helpers.js

```diff
@@ -562,8 +620,9 @@ export function updateSchemaAndData(
   // Update the schema with any fields that are now hidden because of the data change
   newSchema = setHiddenFields(newSchema, uiSchema, formData);
 
-  // Update the schema with any general updates based on the new data
-  newSchema = updateSchemaFromUiSchema(newSchema, uiSchema, formData);
+  // Update the uiSchema and  schema with any general updates based on the new data
+  const newUiSchema = updateUiSchema(newSchema, uiSchema, formData);
```

Comment:
still trying to decide if we should have this part of the normal flow or not. It's extra processing that happens on every form data change (every character change). It adds less than one millisecond of processing, but will scan your entire schema everytime

Reply:
Maybe we could add a timestamp (last updated) to the uiSchema? Or schema?

Reply:
I did that for local testing, but it would a bit too much logging for a standard log. Like I said, it gets called on every form data change. In my testing, it was less than 1ms every time it ran